packages/controls/src/controls/typography/schema.ts

```diff
@@ -0,0 +1,30 @@
+import { z } from 'zod'
+
+import { responsiveValue } from '../../common/schema'
+
+export function typography<C extends z.ZodTypeAny>(color: C) {
+  return z.object({
+    id: z.string().optional(),
+    style: responsiveValue(
+      z.object({
+        fontFamily: z.string().nullable().optional(),
+        lineHeight: z.number().nullable().optional(),
+        letterSpacing: z.number().nullable().optional(),
+        fontWeight: z.number().nullable().optional(),
+        textAlign: z.string().nullable().optional(),
+        uppercase: z.boolean().nullable().optional(),
+        underline: z.boolean().nullable().optional(),
+        strikethrough: z.boolean().nullable().optional(),
+        italic: z.boolean().nullable().optional(),
+        fontSize: z
+          .object({
+            value: z.number().nullable(),
+            unit: z.string().nullable(),
+          })
+          .nullable()
+          .optional(),
+        color: color.optional(),
+      }),
+    ),
+  })
+}
```

Comment:
Zod schema for https://github.com/makeswift/makeswift/pull/781/files#diff-b70dc315c7902612240b31f74d787d26f92c109d2147374ee57dbd6d528da820L3, except that this schema is for a single typography object vs an array.